ixp_tracker/data_lookup.py

```diff
@@ -0,0 +1,68 @@
+import importlib
+import logging
+from datetime import datetime
+from typing import Protocol
+
+logger = logging.getLogger("ixp_tracker")
+
+
+class ASNGeoLookup(Protocol):
+
+    def get_iso2_country(self, asn: int, as_at: datetime) -> str:
+        pass
+
+    def get_status(self, asn: int, as_at: datetime) -> str:
+        pass
+
+    def get_asns_for_country(self, country: str, as_at: datetime) -> list[int]:
+        pass
+
+    def get_routed_asns_for_country(self, country: str, as_at: datetime) -> list[int]:
+        pass
+
+
+class ASNCustomerLookup(Protocol):
+
+    def get_customer_asns(self, asns: list[int], as_at: datetime) -> list[int]:
+        pass
+
+
+class MANRSParticipantsLookup(Protocol):
+
+    def get_manrs_participants(self, as_at: datetime) -> list[int]:
+        pass
+
+
+class AdditionalDataSources(ASNGeoLookup, ASNCustomerLookup, MANRSParticipantsLookup):
```

Comment:
This is the key change here. Clients now need to provide an implementation of AdditionalDataSources (though the lib provides a default no-op implementation if the client doesn't want to provide one) rather than a dict with the different lookups. I've kept the individual Protocols as I think it helps document what the separate data sources are, and means we can type the lookup appropriately where it's used in the lib. And I've extracted everything relevant to that lookup into this file to group it better (and again provide a better place to document it all).